dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/RouteControllerTest.java

```diff
@@ -59,13 +64,41 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.transaction.annotation.Transactional;
 import org.springframework.web.client.RestTemplate;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 
 @Transactional
 class RouteControllerTest extends DhisControllerIntegrationTest {
 
+  private static GenericContainer<?> mockServerContainer;
+
   @Autowired private RouteService service;
 
   @Autowired private ObjectMapper jsonMapper;
+  private MockServerClient mockServerClient;
```

Comment:
I assume you are mocking a service that the routing somehow wants to talk with but that doesn't get clear in any why from the code. Maybe find some names that don't tell me what it is on a technical level (I can see that in the class name) but what the instance is doing, what role it has.